datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -349,9 +349,10 @@ impl PhysicalExpr for InListExpr {
     }
 
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
+        let num_rows = batch.num_rows();
         let value = self.expr.evaluate(batch)?;
         let r = match &self.static_filter {
-            Some(f) => f.contains(value.into_array(1)?.as_ref(), self.negated)?,
+            Some(f) => f.contains(value.into_array(num_rows)?.as_ref(), self.negated)?,
             None => {
                 let value = value.into_array(batch.num_rows())?;
```

Comment:
Maybe we can use the `let value = value.into_array(batch.num_rows())?;` for both arms here

Reply:
Good idea -- done in 7cde104cf